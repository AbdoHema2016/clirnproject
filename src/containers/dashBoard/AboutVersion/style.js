import {StyleSheet} from 'react-native';
import {Layout} from '../../../utilities';

const {LAYOUT_CONSTRAINTS, HEX_COLOR_CODES, FONTS} = Layout;

export const Style = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: HEX_COLOR_CODES.WHITE,
  },
  appTitle: {
    marginTop: LAYOUT_CONSTRAINTS.WINDOW_HEIGHT * 0.13,
    fontSize: 24,
    fontWeight: '600',
    color: HEX_COLOR_CODES.BLACK,
    fontFamily: FONTS.SOFIA_MEDIUM,
    marginLeft: 14,
  },
  info: {
    marginTop: LAYOUT_CONSTRAINTS.WINDOW_HEIGHT * 0.04,
    fontSize: 18,
    fontWeight: '500',
    color: HEX_COLOR_CODES.GRAY,
    fontFamily: FONTS.FUTURA,
    marginLeft: 14,
  },
});
