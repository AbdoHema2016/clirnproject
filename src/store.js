import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
