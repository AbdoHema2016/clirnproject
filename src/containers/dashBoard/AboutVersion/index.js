import React, {useState, useEffect, useCallback} from 'react';
import {View} from 'react-native';
import CLabel from '../../../components/cLabel';
import {Style} from './style';
import {DATEFORMATS, getTestIdentifiers, testIds} from '../../../utilities';
import {translate} from '../../../Localization';

import AsyncStorage, {AsyncConstants} from '../../../utilities/AsyncStorage';
import {version} from '../../../../package.json';
import dayjs from 'dayjs';
const AboutVersion = () => {
  const [installTime, setInstallTime] = useState('');
  useEffect(() => {
    getInstallTime();
  }, [getInstallTime]);

  const getInstallTime = useCallback(async () => {
    try {
      let storageInstallTime = await AsyncStorage.getItemFromStorage(
        AsyncConstants.APP_INSTALL_TIME,
      );
      let formattedDate = formatDate(storageInstallTime);
      setInstallTime(formattedDate);
      return installTime;
    } catch {
      return '';
    }
  }, [installTime]);
  return (
    <View style={Style.container}>
      <CLabel style={Style.appTitle} text={translate('ABOUT_APP.APP_TITLE')} />

      <CLabel
        style={Style.info}
        text={translate('ABOUT_APP.INSTALL_DATE') + installTime}
      />
      <CLabel
        style={Style.info}
        {...getTestIdentifiers(testIds.appBuild)}
        text={translate('ABOUT_APP.VERSION') + version.split('-')[0]}
      />
    </View>
  );
};
const formatDate = (date) => {
  let newDate = dayjs(date).format(DATEFORMATS.ABOUT_VERSION);
  return newDate;
};
export default AboutVersion;
