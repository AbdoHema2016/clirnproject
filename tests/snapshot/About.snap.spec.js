import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import AboutVersion from '../../src/containers/dashBoard/AboutVersion';
jest.mock('@react-navigation/native', () => ({
  withNavigationFocus: (component) => component,
}));

jest.useFakeTimers();

describe('Benefits screen', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
    },
  };
  it('Benefits OTP', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AboutVersion {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
