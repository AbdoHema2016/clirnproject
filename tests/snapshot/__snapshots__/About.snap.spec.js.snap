// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefits screen Benefits OTP 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#2B342C",
          "fontFamily": "SofiaPro-Medium",
          "fontSize": 24,
          "fontWeight": "600",
          "marginLeft": 14,
          "marginTop": 173.42000000000002,
        },
      ]
    }
  >
    [missing "en.ABOUT_APP.APP_TITLE" translation]
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 18,
          "fontWeight": "500",
          "marginLeft": 14,
          "marginTop": 53.36,
        },
      ]
    }
  >
    [missing "en.ABOUT_APP.INSTALL_DATE" translation]
  </Text>
  <Text
    accessibilityLabel="appBuild"
    style={
      Array [
        Object {
          "color": "#8D8C91",
          "fontFamily": "Futura",
          "fontSize": 18,
          "fontWeight": "500",
          "marginLeft": 14,
          "marginTop": 53.36,
        },
      ]
    }
    testID="appBuild"
  >
    [missing "en.ABOUT_APP.VERSION" translation]1.1.22
  </Text>
</View>
`;
